app/uploaders/whitehall_uploader.rb

```diff
@@ -11,4 +11,12 @@ def relative_path
   def assets_protected?
     false
   end
+
+  def asset_params
+    {
+      assetable_id: model.id,
+      asset_variant: version_name ? Asset.variants[version_name] : Asset.variants[:original],
+      assetable_type: model.class.to_s,
+    }.deep_stringify_keys
```

Comment:
I toyed with the idea of creating a model for this, e.g. `AssetReference`, so that I can store the reference in the database. That would enable me to just pass the reference ID to the worker, instead of all the assetable reference data.